lib/tls/tls.cc

```diff
@@ -587,8 +598,9 @@ ssize_t tls_connection_send(Timeout *t,
 					  thread_sleep(&shortSleep);
 					  t->elapse(shortSleep.elapsed);
 				  }
-				  if (!t->may_block())
+				  else
```

Comment:
Are you sure about this? The line above may have reached the timeout condition.

Reply:
Right. The two ifs in a row are correct but this needs a comment.